Add multibench support to all-in-one json

rafaelfolco opened the pull request:
Add benchmarks block to identify bench ids.
The blockbreaker utility extracts the benchmark
from the runfile and processes multibench config.

This requires additional changes on the crucible
code to fully support 'benchmarks' block from the
json runfile.

rafaelfolco committed:
Add multibench support to all-in-one json

Add benchmarks block to identify bench ids. The blockbreaker
utility extracts the benchmark from the runfile and processes
multibench config.

The mv-params for multibench is extracted from the 'benchmarks'
block. To keep backwards compatibility with the single bench mode,
it is still possible to define the mv-params block from the top
level of the JSON file.

For both single bench and multibench on the new schema, use
blockbreaker w/ --config mv-params and --benchmark <benchmark name>.
For single bench on the old schema, use --config mv-params and
--index <index number>.

This requires additional changes on the crucible code to fully
support 'benchmarks' block from the json runfile, and then
support it from the crucible cli.

k-rister commented:
> @k-rister could you restore self-hosted runners please?

done

rafaelfolco commented on util/JSON/schema.json:
```diff
@@ -10,11 +10,20 @@
                 }
             }
         },
+        "benchmarks": {
+            "type": "array",
+            "items": {
+                "minItems": 1,
+                "uniqueItems": true,
+                "type": "object"
+            }
+        },
```
Good point. The other blocks are verified by multiplex. The `benchmarks` block is an abstraction level of CLI and it is not validated anywhere else. Will add this.

HughNhan reviewed (dismissed):
If it works it works!

rafaelfolco commented:
oops found one more dup that needs to be resolved.

rafaelfolco commented:
@k-rister pls review. I modified the schema by copying the mv-params block inside the benchmarks block.  With this change, you can get a mv-params from the json with blockbreaker.py --json all-in-one.json --config mv-params --benchmark uperf. 
Note: benchmarks block was included already in a previous change, so I removed from this one.
